osu.Game/Scoring/ScoreManager.cs

```diff
@@ -72,8 +76,31 @@ protected override ScoreInfo CreateModel(ArchiveReader archive)
             }
         }
 
+        private readonly Dictionary<string, User> previouslyLookedUpUsernames = new Dictionary<string, User>();
```

Comment:
this is permanent memory usage.

you want to be using https://github.com/ppy/osu/blob/master/osu.Game/Database/UserLookupCache.cs if anything.

Reply:
`UserLookupCache` only accepts user IDs right now. It could be modified to take in a string instead of int, but it uses a `GetUsersRequest` to fetch data from online, and that API call only accepts user IDs as a parameter.

Made a new class with similar implementation to `UserLookupCache` (meaning I copied half of it) that gets an ID from username, it uses `GetUserRequest` for each user instead of doing them in a batch. I feel like I'm missing an more sensible way though.